Fix British English Date Format

rom-1 opened the pull request:
The default Date Format for British English is actually not like the american one (M/D/Y), but more like the same as the rest of Europe (D/M/Y). As FreshRSS offers a translation file explicitly for American English (en-US), this one should be considered as Oxford/British english.

rom-1 commented:
whoops, some checks failed? Sorry, did not want to break anything...

Alkarex commented:
@rom-1 Don't worry about the Docker tests, the important one for this type of PR is Travis

Alkarex commented:
Is the dot after the day intentional?

![image](https://user-images.githubusercontent.com/1008324/103479095-8a5d4400-4dcb-11eb-8836-e93df95ae9e9.png)

Cf. https://englishlessonsbrighton.co.uk/how-to-write-dates-british-american-english/

Frenzie commented:
@Alkarex Yeah, but I fixed the Docker test. ;-)

Alkarex commented:
By the way, for another PR if someone could take it, I have just noticed that the layout is slightly broken with most themes for longer en-US dates:
![image](https://user-images.githubusercontent.com/1008324/103479155-0b1c4000-4dcc-11eb-8cda-3a46f2669d50.png)

rom-1 commented:
> Is the dot after the day intentional?
> 
> ![image](https://user-images.githubusercontent.com/1008324/103479095-8a5d4400-4dcb-11eb-8836-e93df95ae9e9.png)
> 
> Cf. https://englishlessonsbrighton.co.uk/how-to-write-dates-british-american-english/

It is not, good find!

rom-1 commented:
> By the way, for another PR if someone could take it, I have just noticed that the layout is slightly broken with most themes for longer en-US dates:
> ![image](https://user-images.githubusercontent.com/1008324/103479155-0b1c4000-4dcc-11eb-8cda-3a46f2669d50.png)

the en-US dates are broken anyway, as US does not have 24h time format -> 12h am/pm

Frenzie commented on app/i18n/es/gen.php:
```diff
@@ -72,8 +72,8 @@
 		'february' => 'febrero',
-		'format_date' => '%s j\\<\\s\\u\\p\\>S\\<\\/\\s\\u\\p\\> Y',	// TODO - Translation
-		'format_date_hour' => '%s j\\<\\s\\u\\p\\>S\\<\\/\\s\\u\\p\\> Y \\a\\t H\\:i',	// TODO - Translation
+		'format_date' => 'j\\. %s Y',
+		'format_date_hour' => 'j\\. %s Y \\a\\t H\\:i',
```
```suggestion
		'format_date' => 'j %s Y',
		'format_date_hour' => 'j %s Y \\a\\t H\\:i',
```

Frenzie commented on app/i18n/en/gen.php:
```diff
@@ -72,8 +72,8 @@
 		'february' => 'February',
-		'format_date' => '%s j\\<\\s\\u\\p\\>S\\<\\/\\s\\u\\p\\> Y',
-		'format_date_hour' => '%s j\\<\\s\\u\\p\\>S\\<\\/\\s\\u\\p\\> Y \\a\\t H\\:i',
+		'format_date' => 'j\\ %s Y',
+		'format_date_hour' => 'j\\ %s Y \\a\\t H\\:i',
```
That \\ isn't for anything anymore, is it?
```suggestion
		'format_date' => 'j %s Y',
		'format_date_hour' => 'j %s Y \\a\\t H\\:i',
```

rom-1 replied:
hope so... :D

Frenzie commented on app/i18n/he/gen.php:
```diff
@@ -72,8 +72,8 @@
 		'february' => 'Feb',
-		'format_date' => '%s j\\<\\s\\u\\p\\>S\\<\\/\\s\\u\\p\\> Y',	// TODO - Translation
-		'format_date_hour' => '%s j\\<\\s\\u\\p\\>S\\<\\/\\s\\u\\p\\> Y \\a\\t H\\:i',	// TODO - Translation
+		'format_date' => 'j\\ %s Y',
+		'format_date_hour' => 'j\\ %s Y \\a\\t H\\:i',
```
```suggestion
		'format_date' => 'j %s Y',
		'format_date_hour' => 'j %s Y \\a\\t H\\:i',
```

Frenzie commented on app/i18n/ru/gen.php:
```diff
@@ -72,8 +72,8 @@
-		'format_date' => '%s j\\<\\s\\u\\p\\>S\\<\\/\\s\\u\\p\\> Y',	// TODO - Translation
-		'format_date_hour' => '%s j\\<\\s\\u\\p\\>S\\<\\/\\s\\u\\p\\> Y \\a\\t H\\:i',	// TODO - Translation
+		'format_date' => 'j\\. %s Y',
+		'format_date_hour' => 'j\\. %s Y \\a\\t H\\:i',
```
```suggestion
		'format_date' => 'j %s Y',
		'format_date_hour' => 'j %s Y \\a\\t H\\:i',
```

Frenzie commented on app/i18n/tr/gen.php:
```diff
@@ -72,8 +72,8 @@
 		'february' => 'Şub',
-		'format_date' => '%s j\\<\\s\\u\\p\\>S\\<\\/\\s\\u\\p\\> Y',	// TODO - Translation
-		'format_date_hour' => '%s j\\<\\s\\u\\p\\>S\\<\\/\\s\\u\\p\\> Y \\a\\t H\\:i',	// TODO - Translation
+		'format_date' => 'j\\. %s Y',
+		'format_date_hour' => 'j\\. %s Y \\a\\t H\\:i',
```
```suggestion
		'format_date' => 'j %s Y',
		'format_date_hour' => 'j %s Y \\a\\t H\\:i',
```

rom-1 commented:
ok, so I added more commits, removing the dot, and also fixed the american time (24h -> 12h). As I have almost no experience using GIT, just my simple question: Is there anything else I need to do right now to commit all these changes? Or is everything alright from my side, and you guys just have to approve it?

Alkarex commented:
@rom-1 You did it just right, thanks. Next steps are review + test + merge from our side.

Alkarex reviewed (changes requested):
There is a little warning on the Travis automatic test:
https://travis-ci.org/github/FreshRSS/FreshRSS/jobs/752642159

Alkarex commented:
Because `format_date` is now identical in `en` than some other languages, the corresponding "ignore" files must be updated

rom-1 commented:
> Because `format_date` is now identical in `en` than some other languages, the corresponding "ignore" files must be updated
I could simply copy Frenzie's work and add 2 lines ('gen.date.format_date', 'gen.date.format_date_hour',) to the array of every cli/i18n/ignore/xx.php file. But while this might be the correct way to fix it, I would have no idea what I am doing. So maybe I am not the best person changing this?

aledeg commented:
You can do it with the help of the make command

rom-1 commented:
> You can do it with the help of the make command

you are by far overestimating my skills! :)
I have never used make in the past nor have a linux/unix machine at hand. 

I manually added Frenzie's 2 lines to the array of every language file except of the en.php one. As stated, I have no idea what I am doing here, but maybe this will fix the travis checks.

Frenzie commented:
There's no need to add it to everything, just the ones that are supposed to be identical to the source. Which right now is only fr, nl and pl I think? You need to look at the checks to see what's actually failing: https://travis-ci.org/github/FreshRSS/FreshRSS/jobs/752642159

I think it'd be better to revert the ignore commits for the other languages. I can probably do that if desired, but it'd have to wait a while.

rom-1 commented:
ok, reverted all commits except fr, nl and pl, as suggested by Frenzie

Frenzie commented on cli/i18n/ignore/pl.php:
```diff
@@ -1,6 +1,8 @@
 <?php
 
 return array(
+	'gen.date.format_date',
+	'gen.date.format_date_hour',
```
We'll have to wait a bit for Travis, but I suspect/hope it'll complain this isn't alphabetical. ;-)
